tests/unit/test_charm.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+"""Test default charm events such as upgrade charm, install, etc."""
+
+import unittest
+from unittest.mock import PropertyMock, patch
+
+import ops.testing
+from charm import SlurmdbdCharm
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+ops.testing.SIMULATE_CAN_CONNECT = True
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self) -> None:
+        self.harness = Harness(SlurmdbdCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    @unittest.expectedFailure
+    def test_install_success(self) -> None:
+        """Test that slurmdbd can successfully be installed.
+
+        Notes:
+            This method is expected to fail due to the 'version' file missing.
+        """
+        self.harness.charm.on.install.emit()
+        self.assertEqual(
+            self.harness.charm.unit.status, ActiveStatus("slurmdbd successfully installed")
+        )
+
+    @unittest.expectedFailure
+    @patch("slurm_ops_manager.SlurmManager.install")
+    def test_install_fail(self, install) -> None:
+        """Test that slurmdbd install fail handler works.
+
+        Notes:
+            This method is expected to fail due to the 'version' file missing.
+        """
```

Comment:
Naming is confusing as well. The test expects to fail but it is checking if the fail works

Reply:
The reason why this test is decorated with `@unittest.expectedFailure` is because I am expecting it to fail due to the 'version' file missing ([see James's comment](https://github.com/omnivector-solutions/slurmdbd-operator/pull/1/files/f97a529f18f374933632bff1a5b5d5280c0a89f3#r1067663862)). Once we hammer out what we would like to do about the version file, I will remove the `@unittest.expectedFailure` decorator. 

With the decorator removed, I am hoping to test that the following branch is executing correctly in `_on_install` inside `charm.py`:

https://github.com/omnivector-solutions/slurmdbd-operator/blob/78eb66ee34dfadd86a9e93f76a60e174fb17853b/src/charm.py#L95-L98

My mistake, I should have communicated this in the pull request description.